xarray/core/common.py

```diff
@@ -1029,6 +1029,9 @@ def where(
         self: T_DataWithCoords, cond: Any, other: Any = dtypes.NA, drop: bool = False
     ) -> T_DataWithCoords:
         """Filter elements from this object according to a condition.
+        
+        Returns elements from 'DataArray', where 'cond' is True, 
+        otherwise fill in 'other' or 'drop'.
```

Comment:
Maybe better to write "replace the rest by 'other' (default: NaN)"?

'or drop" is extremely vague, you could understand it as an alternative to 'other'. Not sure to what extent you should explain what drop does here vs. in the argument list.
Could be something like "drop completely masked elements along all dimensions", even though we don't talk about masking here... Maybe someone finds a better wording...

Reply:
What do you think about this?

Returns elements from 'DataArray', where 'cond' is True. The rest is replaced by 'other' (default: NaN) or dropped if drop=True.

Reply:
This sounds good, except we need to elaborate a bit on `drop`: A common confusion when using `where(..., drop=True)` with multi-dimensional arrays is that `cond` has to be all-`False` along any dimension for anything to be dropped, so we should definitely mentions that in the parameter description (have a look at the examples in the docstring if you're wondering what I'm talking about)

Reply:
> Returns elements from 'DataArray', where 'cond' is True. The rest is replaced by 'other' (default: NaN) or dropped if drop=True."

FWIW I think @Riulinchen 's proposal is much better than the current description, and we could merge that.

(agree we could also add more on how other dims are dropped, but don't need to gate the improvement on that?)

Reply:
sure, we can of course do that in a different PR (we just have to remember to actually do so)